inferno/tests/test_dataset.py

```diff
@@ -49,6 +50,175 @@ def test_inconsistent_lengths(self, get_len, data):
             get_len(data)
 
 
+class TestNetWithoutY:
+
+    net_fixture_params = [
+        {'classification': True, 'batch_size': 1},
+        {'classification': False, 'batch_size': 1},
+        {'classification': True, 'batch_size': 2},
+        {'classification': False, 'batch_size': 2},
+    ]
+
+    @pytest.fixture
+    def net_cls_1d_clf(self):
+        class MyModule(nn.Module):
+            def __init__(self):
+                super(MyModule, self).__init__()
+                self.dense = nn.Linear(1,1)
+            def forward(self, X):
+                return F.softmax(self.dense(X.float()))
```

Comment:
Maybe it's not realistic, but for these tests to run, we don't need the final nonlinearity. If you drop it, clf and reg would use the same module and you could safe some almost duplicate code.